syncd/SaiSwitch.cpp

```diff
@@ -1030,6 +1030,20 @@ void SaiSwitch::onPostPortCreate(
          */
 
         redisSetDummyAsicStateForRealObjectId(rid);
+
+	/*
+         * The ports which is without QUEUE related configurations still created
+         * related SCHEDULER_GROUPs in the onPostPortCreate() during the syncd
+         * startup in some platform - e.g. montara. But these created objects
+	 * did NOT be recorded in the `COLDVIDS` and `m_warmBootDiscoveredVids`
+         * set. While swss is perform warmstart, the `compareViews()` would be
+         * called before apply the view, and it would detect the differences
+         * between the current view and temp view. Then syncd would remove these
+         * schedulers groups in the finalize. Therefore, add to record these
+         * object into the m_warmBootDiscoveredVids
+         */
+        sai_object_id_t vid = m_translator->translateRidToVid(rid, m_switch_vid);
+        m_warmBootDiscoveredVids.insert(vid);
```

Comment:
not sure why this is needed here, why m_warmBootDiscoveredVids is used at all, since onPostPortCreate is called AFTER cold boot or warm boot was performed already, so all discovered objects were already discovered during "create switch" at that phase. i dont quite understand description here, also you don't know whether onPostPortCreate which is called here is after cold boot or after warm boot, but yet  you always add vids to m_warmBootDiscoveredVids

this translation is not needed here, if you are missing some vids, then it means problem is somewhere else

Reply:
Thanks for reply. The case that I encountered was to perform warmstart swss only, not syncd. And the onPostPortCreate() would not be called again. And it causes some vids miss that which are created during previous onPostPortCreate(), because there is no records for this vids.